message/message-http/src/main/java/com/mastercard/test/flow/msg/http/HttpMsg.java

```diff
@@ -248,16 +248,24 @@ protected static boolean isHeaderField( String field ) {
 	}
 
 	/**
+	 * Creates a header field name. Note that
+	 * <a href="https://www.rfc-editor.org/rfc/rfc9110.html#name-field-names">header
+	 * fields are case-insensitive</a>. Accordingly, this method will convert the
+	 * supplied name to lower-case before adding the sigils that identify it as a
+	 * header field.
+	 *
 	 * @param name The desired header name
 	 * @return The value to pass as the field name in {@link #set(String, Object)}
 	 *         and {@link #get(String)}
 	 */
 	public static String header( String name ) {
-		return HEADER_PRESUFIX + name + HEADER_PRESUFIX;
+		return HEADER_PRESUFIX + name.toLowerCase() + HEADER_PRESUFIX;
```

Comment:
This is the key change - we now convert all header names to lower case when they get added to the message data model.

Reply:
Why the prefix/suffix on headers? I'm not familiar with it being part of the specification.